### PR TITLE
Fix terminating Layer and isQueryable

### PR DESCRIPTION
**Pull request by edellucien**
# How to

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [ ] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] The title means something for a human being.
- [x] The title contains [WIP] if it's necessary.
- [ ] Labels applied. if it's a release? a hotfix?
- [ ] Tests added.

### CODE REVIEW

File: src/common/layers/Layer.js [hunk 1]
```diff
@@ -33,7 +34,7 @@ export default class Layer extends Observable {
    * @param {Object} [options.properties={}] Application-specific layer properties.
    * @param {boolean} [options.visible=true] If true this layer is visible on the map.
    * @param {boolean} [options.isBaseLayer=false] If true this layer is a baseLayer.
-   * @param {boolean} [options.isQueryable=false] If true feature information can be queried by the react-spatial LayerService. Default is undefined, but resulting to true if not strictly set to false.
+   * @param {boolean} [options.isQueryable=undefined] If true feature information can be queried by the react-spatial LayerService. Default is undefined, but resulting to true if not strictly set to false.
```
Comment (oterral) [inline src/common/layers/Layer.js:37]: no it should be false by default

File: src/common/layers/Layer.js [hunk 2]
```diff
@@ -117,13 +119,29 @@ export default class Layer extends Observable {
     this.terminate();
     /** @ignore */
     this.map = map;
+    if (!this.map || !this.olLayer) {
+      return;
+    }
+
+    this.olListenersKeys.push(
+      this.map.getLayers().on('remove', (evt) => {
```
Comment (oterral) [inline src/common/layers/Layer.js:127]: you are in the common layer , there should not be reference to mapbox or ol function, move it to src/ol/layers

File: src/common/layers/Layer.js [hunk 3]
```diff
@@ -117,13 +119,29 @@ export default class Layer extends Observable {
...
    * Terminate what was initialized in init function. Remove layer, events...
    */
   // eslint-disable-next-line class-methods-use-this
-  terminate() {}
+  terminate() {
+    unByKey(this.olListenersKeys);
+    if (this.map && this.olLayer) {
+      this.map.removeLayer(this.olLayer);
+    }
+  }
```
Comment (oterral) [inline src/common/layers/Layer.js:144]: same here

File: src/common/layers/Layer.js (hunk 2 above)
Comment (edellucien) [reply src/common/layers/Layer.js:127]: Obviously you are right :+1: I correct it

File: package.json [hunk 4]
```diff
@@ -36,6 +36,12 @@
     "typeface-lato": "^0.0.75",
     "uuid": "^8.1.0"
   },
+  "peerDependencies": {
+    "mapbox-gl": "^1.10.1",
+    "ol": "^6.3.1",
+    "react": "^16.13.1",
+    "react-dom": "^16.13.1"
```
Comment (oterral) [inline package.json:43]: react and react-dom  can be in dependecies , there is no react stuff in the library only for the documentation

File: src/ol/layers/Layer.js [hunk 5]
```diff
@@ -13,7 +13,7 @@ import LayerCommon from '../../common/layers/Layer';
  * @param {Object} [options.properties={}] Application-specific layer properties.
  * @param {boolean} [options.visible=true] If true this layer is the currently visible layer on the map.
  * @param {boolean} [options.isBaseLayer=false] If true this layer is a baseLayer.
- * @param {boolean} [options.isQueryable=false] If true feature information can be queried by the react-spatial LayerService. Default is undefined, but resulting to true if not strictly set to false.
+ * @param {boolean} [options.isQueryable=undefined] If true feature information can be queried by the react-spatial LayerService. Default is undefined, but resulting to true if not strictly set to false.
```
Comment (oterral) [inline src/ol/layers/Layer.js:16]:
```suggestion
 * @param {boolean} [options.isQueryable=false] If true feature information can be queried by the react-spatial LayerService. Default is undefined, but resulting to true if not strictly set to false.
```

File: src/ol/layers/Layer.js [hunk 6]
```diff
@@ -40,13 +40,28 @@ class Layer extends LayerCommon {
...
+    );
   }
 
   /**
    * Terminate what was initialized in init function. Remove layer, events...
    */
   terminate() {
     unByKey(this.olListenersKeys);
+    if (this.map && this.olLayer) {
+      this.map.removeLayer(this.olLayer);
```
Comment (oterral) [inline src/ol/layers/Layer.js:63]: init doesn't add the layer so terminate shouldn't remove the layer.

**Pull request merged**